eth-providers/src/utils/transactionHelper.ts

```diff
@@ -0,0 +1,69 @@
+import { BigNumber, BigNumberish } from 'ethers';
+
+type TxConsts = {
+  storageByteDeposit: BigNumberish;
+  txFeePerGas: BigNumberish;
+};
+
+const bigNumDiv = (x: BigNumber, y: BigNumber) => {
+  const res = x.div(y);
+  return res.mul(y) === x ? res : res.add(1);
+};
+
+export const calcEthereumTransactionParams = (
+  data: {
+    gasLimit: BigNumberish;
+    storageLimit: BigNumberish;
+    validUntil: BigNumberish;
+  } & TxConsts
+): {
+  txGasPrice: BigNumber;
+  txGasLimit: BigNumber;
+} => {
+  const gasLimit = BigNumber.from(data.gasLimit);
+  const storageLimit = BigNumber.from(data.storageLimit);
+  const validUntil = BigNumber.from(data.validUntil);
+  const storageByteDeposit = BigNumber.from(data.storageByteDeposit);
+  const txFeePerGas = BigNumber.from(data.txFeePerGas);
+
+  const blockPeriod = validUntil.div(30);
+  const storageEntryLimit = storageLimit.div(64);
+  const storageEntryDeposit = storageByteDeposit.mul(64);
+  const txGasPrice = txFeePerGas.add(blockPeriod.shl(16)).add(storageEntryLimit);
+
+  const txGasLimit = storageEntryLimit.mul(storageEntryDeposit).div(txFeePerGas).add(gasLimit);
```

Comment:
storageEntryLimit = 3
storageEntryDeposit = 4
txFeePerGas = 5
```
3 * 4 / 5 = 2
4 / 5 * 3 = 0
```
This will cause inconsistent calculations